apiserver/facades/agent/credentialvalidator/credentialvalidator.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+package credentialvalidator
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/loggo"
+	"gopkg.in/juju/names.v2"
+
+	"github.com/juju/juju/apiserver/common"
+	"github.com/juju/juju/apiserver/facade"
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/state"
+	"github.com/juju/juju/state/watcher"
+)
+
+var logger = loggo.GetLogger("juju.api.credentialvalidator")
+
+// CredentialValidator defines the methods on credentialvalidator API endpoint.
+type CredentialValidator interface {
+	ModelCredentials(params.Entities) (params.ModelCredentialResults, error)
+	WatchCredential(params.Entities) (params.NotifyWatchResults, error)
+}
+
+type CredentialValidatorAPI struct {
+	backend   Backend
+	resources facade.Resources
+}
+
+var _ CredentialValidator = (*CredentialValidatorAPI)(nil)
+
+// NewCredentialValidatorAPI creates a new CredentialValidator API endpoint on server-side.
+func NewCredentialValidatorAPI(st *state.State, resources facade.Resources, authorizer facade.Authorizer) (*CredentialValidatorAPI, error) {
```

Comment:
New facades should be constructed with a facade context. Using this constructor is legacy